### PR TITLE
docs: README.md with documentation for testing purposes

### PR DESCRIPTION
**Pull request by yashksaini-coder**
Fixes:- #2 

/claim #2 

Enhance README.md with comprehensive documentation for `fake-anthropic` library, including setup instructions, integration examples, and effective testing strategies.

Documentation updates:

* Added an introduction to explain the purpose of the `@tscircuit/fake-anthropic` library and its benefits for testing.
* Provided detailed instructions for setting up the test environment, including required dependencies and configuration.
* Included a guide for integrating `fake-anthropic` into tests with an example test case.
* Added a section on mocking external services using `fake-anthropic` with a sample implementation.
* Suggested effective testing strategies to ensure robust and maintainable tests.

### CODE REVIEW

**Comment by yashksaini-coder** [comment]
@seveibar can you please review the PR

File: README.md [hunk 1]
```diff
@@ -1,5 +1,211 @@
...
+   ```
+
+## Effective testing strategies
+
+Here are some effective testing strategies to consider when using `fake-anthropic`:
+
+* Use snapshot testing to ensure that the responses from the fake Anthropic API match the expected output. This can help you catch any unexpected changes in the API responses.
+* Isolate your tests by using a separate test database for each test run. This can help you avoid conflicts and ensure that your tests are independent of each other.
+* Use middleware to inject dependencies, such as the database client, into your test context. This can help you keep your tests clean and maintainable.
+* Mock external services to simulate different scenarios and edge cases. This can help you test your application's behavior under various conditions.
```
Comment (imrishabh18) [inline README.md]: I don't think these are needed or related to the `fake-anthropic` testing

File: README.md [hunk 2]
```diff
@@ -1,5 +1,211 @@
...
+         "output_tokens": 0,
+       },
+     }
+     `);
+   });
+   ```
+
+## Mocking external services using `fake-anthropic`
+
+To mock external services using `fake-anthropic`, you can create a fake implementation of the Anthropic API and use it in your tests. This allows you to simulate interactions with the Anthropic service without making real API calls.
```
Comment (imrishabh18) [inline README.md]: We don't want to use the term `mock`, it should be replaced with `fake`, `faking` etc

**Review by imrishabh18 — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (yashksaini-coder) [reply README.md]: Replaced with `Fake` @seveibar

File: README.md (hunk 1 above)
Comment (yashksaini-coder) [reply README.md]: Removed the unecessary points @seveibar

File: README.md [hunk 3]
```diff
@@ -1,5 +1,201 @@
...
+       axios,
+     };
+   };
+   ```
+
+## Integrating `fake-anthropic` in tests
+
+To integrate `fake-anthropic` in your tests, follow these steps:
+
+1. Create a test for the Anthropic SDK usage in `tests/usage/usage1-anthropic-sdk.test.ts`:
```
Comment (imrishabh18) [inline README.md]: I think according to the issue, it wants to show how to use this `fake-anthropic` in other application tests.So, for that we need to show an example of using the `/dist/bundle` of the `fake-anthropic` which gets generated

cc: @seveibar

**Review by imrishabh18 — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (yashksaini-coder) [reply README.md]: okh I got it @imrishabh18

File: README.md [hunk 4]
```diff
@@ -1,5 +1,201 @@
...
+       "module": "ESNext",
+       "moduleResolution": "bundler",
+       "strict": true,
+       "esModuleInterop": true,
+       "resolveJsonModule": true
+     },
+     "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
+     "exclude": ["node_modules"]
+   }
+   ```
```
Comment (seveibar) [inline README.md]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -1,5 +1,201 @@
...
+     });
+
+     afterEach(async () => {
+       await server.stop();
+     });
+
+     return {
+       url,
+       server,
+       axios,
```
Comment (seveibar) [inline README.md]: can we switch to ky using ky.create({ prefixUrl }) here?

File: README.md (hunk 3 above)
Comment (seveibar) [reply README.md]: yes that's correct! @imrishabh18

**Comment by yashksaini-coder** [comment]
@seveibar @imrishabh18 Kindly review the PR

**Comment by yashksaini-coder** [comment]
Hi @seveibar @imrishabh18 any new updates ?

File: README.md [hunk 6]
```diff
@@ -1,5 +1,188 @@
...
+    };
+   ```
+
+## Using `fake-anthropic` in other application tests
+
+To use the `fake-anthropic` bundle in other application tests, follow these steps:
+
+1. Install the `fake-anthropic` package in your application:
+  ```sh
+  npm install @tscircuit/fake-anthropic
```
Comment (imrishabh18) [inline README.md]:
```suggestion
  bun add @tscircuit/fake-anthropic
```

File: README.md [hunk 7]
```diff
@@ -1,5 +1,188 @@
 # @tscircuit/fake-anthropic
 
-- Winterspec for defining the API
-- bun testing
-- Zustand store with zod definition for database state
+## Introduction
```
Comment (imrishabh18) [inline README.md]: I don't think we need both Introduction as well as what is fake-anthropoc?

Both of them are having the same details. Better remove one of them

**Review by imrishabh18 — CHANGES_REQUESTED** [review]

File: README.md (hunk 7 above)
Comment (yashksaini-coder) [reply README.md]: done removed the latter

File: README.md (hunk 6 above)
Comment (yashksaini-coder) [reply README.md]: done change the npm --> bun

**Comment by yashksaini-coder** [comment]
@imrishabh18 hi sir, I have made changes as per requested please review them.

**Review by imrishabh18 — APPROVED** [review]
LGTM! @yashksaini-coder 🚀

**Comment by yashksaini-coder** [comment]
> LGTM! @yashksaini-coder 🚀

Thx for the quick response bro. This will be my first bounty PR of 2025

File: README.md [hunk 8]
```diff
@@ -1,5 +1,184 @@
...
+     },
+    });
+  });
+  ```
+
+## Fake external services using `fake-anthropic`
+
+To Fake external services using `fake-anthropic`, you can create a fake implementation of the Anthropic API and use it in your tests. This allows you to simulate interactions with the Anthropic service without making real API calls.
+
+1. Create a fake implementation of the Anthropic API in `routes/v1/messages/index.ts`:
```
Comment (seveibar) [inline README.md]: i don't understand this section, this isn't how you use this?

File: README.md [hunk 9]
```diff
@@ -1,5 +1,184 @@
...
+
+2. Import and use the `fake-anthropic` bundle in your test file:
+  ```ts
+  import { test, expect } from "bun:test";
+  import { getTestServer } from "tests/fixtures/get-test-server";
+  import { normalizeForSnapshot } from "tests/fixtures/normalizeForSnapshot";
+  import { Anthropic } from "@tscircuit/fake-anthropic/dist/bundle";
+
+  test("integration test with fake-anthropic", async () => {
+    const { url } = await getTestServer();
```
Comment (seveibar) [inline README.md]: the consumer doesn't have `getTestServer`, you have to show them how to use the package- and your usage is wrong...

**Review by seveibar — CHANGES_REQUESTED** [review]
this isn't really how you use this package
